scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -54,8 +61,39 @@ void analyze() {
     @Override
     public void visit(CheckExpressionNode checkExpressionNode) {
         if (checkExpressionNode.checkKeyword().kind().equals(SyntaxKind.CHECKPANIC_KEYWORD)) {
-            scannerContext.getReporter().reportIssue(document, checkExpressionNode.location(),
+            reportIssue(scannerContext, document, checkExpressionNode,
                     CoreRule.AVOID_CHECKPANIC.rule());
         }
     }
+
+    @Override
+    public void visit(ModulePartNode modulePartNode) {
+        modulePartNode.members().forEach(member -> member.accept(this));
+    }
+
+    @Override
+    public void visit(FunctionSignatureNode functionSignatureNode) {
+        functionSignatureNode.parameters().forEach(parameter -> {
+            semanticModel.symbol(parameter).ifPresent(symbol -> {
+                if (semanticModel.references(symbol).size() == 1) {
+                    reportIssue(scannerContext, document, parameter,
+                            CoreRule.UNUSED_FUNCTION_PARAMETERS.rule());
+                }
+            });
+            parameter.accept(this);
+        });
+    }
+
+    @Override
+    public void visit(MethodCallExpressionNode methodCallExpressionNode) {
```

Comment:
Assuming we are using this to test arrow function calls, shall we verify this don't have any adverse effects on "normal" method calls.